### PR TITLE
upgrade e2e and enable firefox

### PR DESCRIPTION
**Pull request by liunate**
###  Features and Bug Fixes

~- [ ] Issue being fixed is referenced.~
~- [ ] Unit test coverage added/updated.~
~- [ ] E2E test coverage added/updated.~
~- [ ] Prop-types added/updated.~
~- [ ] Typings added/updated.~
~- [ ] README updated (API changes, instructions, etc.).~
~- [ ] Changes to dependencies explained.~
~- [ ] Changeset added (run `yarn changeset` locally to add one, follow prompts).~

### Tooling

- [x] Issue being fixed is referenced.
- [x] Changes to dependencies explained.
~- [ ] README instructions updated.~
- [x] A Changeset is __not__ added.

### DONE
- This is fixing cypress firefox issue on git action https://github.com/cypress-io/cypress-docker-images/issues/363 and also #104 
- Cypress container is updated to keep currency with `package.json` `devDependencies`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,18 +52,17 @@ jobs:
...
   e2e:
     runs-on: ubuntu-latest
-    container: cypress/included:5.0.0
     strategy:
       matrix:
-        browser: [chrome] # cypress executable says 'Browser: 'firefox' was not found on your system or is not supported by Cypress.' seems to be defect
+        browser: [chrome, firefox]
+    container:
+      image: cypress/included:6.0.1
+      options: --user 1001
```
Comment (liunate) [inline .github/workflows/ci.yml:60]: Per official cypress git action example for Firefox: https://github.com/cypress-io/github-action#firefox

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -52,18 +52,17 @@ jobs:
...
-      - name: Install dependencies 📦
-        uses: bahmutov/npm-install@v1
-
-      - name: Run tests 🧪
-        run: yarn test:cypress-ci
+      - uses: actions/checkout@v1
+      - uses: cypress-io/github-action@v2.6.1
+        with:
+          start: yarn start
+          browser: ${{ matrix.browser }}
```
Comment (liunate) [inline .github/workflows/ci.yml:68]: This is the suggested git action way to run cypress test on CI by [bamutov](https://github.com/cypress-io/cypress-docker-images/issues/363#issuecomment-729948368)

**Comment by liunate** [comment]
One comment out of this change but I just thinked about at the trigger condition of e2e test https://github.com/focus-trap/focus-trap-react/pull/206/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR6
```yml
```

**Comment by liunate** [comment]
If the change is good I will also port this to the base repo `focus-trap`

**Review by stefcameron — APPROVED** [review]
Fantastic! Thanks for keeping track of the Cypress FF issue. 🎉

**Comment by stefcameron** [comment]
> Do we benefit from running e2e at every commit push so submitter can find out real browser issue much earlier than PR creation? I feel this repo e2e is really fast.

AFAIK, the CI action is configured such that it will run for any PR update (whether you force-push the branch or add another commit) as well as on master for all merges/commits to master.

Are you seeing something different?

**Comment by stefcameron** [comment]
> If the change is good I will also port this to the base repo `focus-trap`

Yes, please do!

**Comment by stefcameron** [comment]
FYI, I have now made the FF e2e tests in CI mandatory, as in must pass before merging.

**Pull request merged**